config/initializers/5_telemetry.rb

```diff
@@ -1,11 +1,117 @@
-# Foreman telemetry global setup.
-telemetry = Foreman::Telemetry.instance
-if SETTINGS[:telemetry] && (Rails.env.production? || Rails.env.development?)
-  telemetry.setup(SETTINGS[:telemetry])
-end
+# Foreman telemetry metrics registration.
+#
+# There are three types of telemetry measurements: counter, gauge, histogram. Each measurement has unique name
+# represented by a symbol, a description and list of tags which are required and mapped to monitoring frameworks
+# which do not support arbitrary tags.
+#
+# Plugins can add own metrics through add_counter_telemetry, add_gauge_telemetry and add_histogram_telemetry.
+#
+Rails.application.config.to_prepare do
```

Comment:
I believe we should rather just `require 'foreman/telemetry'` as
1. I'd like to aim to `/lib` being removed from autloaded paths
2. I don't think reloading telemetry rules is something we want to do or is much usefull